src/zenml/cli/stack_recipes.py

```diff
@@ -0,0 +1,840 @@
+from zenml.cli.stack import stack
+from zenml.console import console
+from zenml.exceptions import GitNotFoundError
+from zenml.io import fileio
+from zenml.logger import get_logger
+from zenml.utils import io_utils
+from zenml.utils.analytics_utils import AnalyticsEvent, track_event
+
+logger = get_logger(__name__)
+
+EXCLUDED_RECIPE_DIRS = [""]
+STACK_RECIPES_GITHUB_REPO = "https://github.com/zenml-io/mlops-stacks.git"
+STACK_RECIPES_REPO_DIR = "zenml_stack_recipes"
+RECIPE_RUN_SCRIPT = "run_recipe.sh"
+RECIPE_DESTROY_SCRIPT = "destroy_recipe.sh"
+RECIPE_DEPLOY_IDENTIFIER = "deploy"
+RECIPE_DESTROY_INDENTIFIER = "destroy"
+SHELL_EXECUTABLE = "SHELL_EXECUTABLE"
+
+
+class LocalStackRecipe:
+    """Class to encapsulate the local stack that can be run from the CLI."""
+
+    def __init__(self, path: Path, name: str) -> None:
+        """Create a new LocalStack instance.
+
+        Args:
+            name: The name of the stack, specifically the name of the folder
+                  on git
+            path: Path at which the stack is installed
+        """
+        self.name = name
+        self.path = path
+
+    @property
+    def needs_manual_user_setup(self) -> bool:
+        """Checks if manual setup is required from the user.
+
+        Returns:
+            True if manual setup required, False else
+        """
+        raise NotImplementedError
+
+    @property
+    def recipes_deploy_bash_script(self) -> str:
+        """Path to the bash script that deploys the recipe.
+
+        Returns:
+            Path to the bash script that deploys the recipe.
+        """
+        return os.path.join(self.path, RECIPE_RUN_SCRIPT)
+
+    @property
+    def recipes_destroy_bash_script(self) -> str:
+        """Path to the bash script that destroys the recipe.
+
+        Returns:
+            Path to the bash script that destroys the recipe.
+        """
+        return os.path.join(self.path, RECIPE_DESTROY_SCRIPT)
+
+    # @property
+    # def executable_python_stack_recipe(self) -> str:
+    #     """Return the Python file for the stack_recipe.
+
+    #     Returns:
+    #         The Python file for the stack_recipe.
+
+    #     Raises:
+    #         RuntimeError: If no runner script is present in the stack_recipe.
+    #         NotImplementedError: If the stack_recipes needs manual user setup.
+    #     """
+    #     if self.needs_manual_user_setup:
+    #         raise NotImplementedError(
+    #             "This recipe currently does not support being run from the "
+    #             "CLI as user specific setup is required. Consult the README.md "
+    #             "of the stack recipe to find out more."
+    #         )
+
+    def is_present(self) -> bool:
+        """Checks if the stack_recipe exists at the given path.
+
+        Returns:
+            True if the stack_recipe exists at the given path, else False.
+        """
+        return fileio.exists(str(self.path)) and fileio.isdir(str(self.path))
```

Comment:
i. think `fileio.isdir(str(self.path))` is enough here. as it would verify if the file exists or not